x-pack/platform/plugins/shared/fleet/server/routes/agent/index.ts

```diff
@@ -123,6 +126,38 @@ export const registerAPIRoutes = (router: FleetAuthzRouter, config: FleetConfigT
       getAgentHandler
     );
 
+  // Migrate
+  router.versioned
+    .post({
+      path: AGENT_API_ROUTES.MIGRATE_PATTERN,
+      security: {
+        authz: {
+          requiredPrivileges: [FLEET_API_PRIVILEGES.AGENTS.ALL],
```

Comment:
could be worth adding a scenario here to ensure privileges are correct https://github.com/elastic/kibana/blob/main/x-pack/test/fleet_api_integration/apis/agents/privileges.ts#L300-L306